src/wovnio/wovnphp/core/WovnLang.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+
+namespace Wovnio\Wovnphp\Core;
+
+class WovnLang
+{
+    private $name;
+    private $code;
+    private $englishName;
+    private $alias;
+    private $active;
+
+    /**
+     * WovnLang constructor.
+     * @param $code string The language code, IETF language tag format.
+     * @param $name string The name of the language, in its local language.
+     * @param $englishName string The name of the language, in English.
+     * @param mixed $alias The alias of this language, defaults to null.
+     * @param bool $active If the language is a valid choice in the current project.
+     */
+    public function __construct($code, $name, $englishName, $alias = null, $active = false)
```

Comment:
I would say this class is doing too much personally. Don't really like the "active" status, and I would make language aliases be managed separately from this class myself.

Methods like "setAlias", "enable" are kind of off. These things should only be getting set once on initialization, so don't really deserve methods for.

Reply:
Yep, I agree the setAlias() and enable() should not be there. These were hacked on when I was working on LangDirectory.

But I still believe it's necessary to keep information such as if the language is in use in the project or the language's alias inside the Lang class. My rationale is as the following:

1. Logically the alias of a language and its validity are information that describes a language
2. It is easier to do `$lang->alias()` than `LangAlias->getAlias($lang)`, like the focus is not broken.

Reply:
It is definitely more convenient, but I think it comes at the expense of code "purity" 😛 

Another operation we need to do is reverse-lookup a language from an alias, and I think there is symmetry with having both of these operations being close together in code.

Reply:
The reverse lookup is actually handled by "getting" the lang. The design is that whenever you want to get a Lang object, you call `WovnLangDirectory->getLang()`, it doesn't matter if you pass in a proper lang code or an alias. And you also get the same Lang object back for the same language.

Reply:
Yeah, I get that WovnLangDirectory is kind of serving both purposes.